### ISSUE TITLE
Add ability for user to send signals to job (e.g. for clean abort)

### ISSUE THREAD

**Issue by fcasson** [prominence-eosc/prominence#81]
In our code we trigger a clean abort by touching a specific file in the run directory.

A mechanism to do this for a running job in prominence would allow for clean early terminations.

**Comment by alahiff** [comment]
I assume this is only for when you want to do something like `prominence delete <job id>`? 

I should be able to have a signal sent to the job (by default SIGTERM), then can wait for a few seconds before forcibly killing the job. If you wil need a copy of the run directory for such cases I'll need to handle this as well.

**Comment by fcasson** [comment]
Yes, It's a way of forcing an early clean termination with restart file output.  We would need the output in this case, and possibly also in the case of normal deleted jobs.

If its done by SIGTERM then we will need some re-engineering on our side, and likely we wouldn't be able to use this feature for while.

[I can add priorities to all my feature requests, if I get permission to add labels]

**Comment by alahiff** [comment]
It doesn't have to be SIGTERM, is there another signal you'd prefer? I could even make it configurable.

Or would you prefer the ability to touch the special file which triggers the clean abort?

**Comment by fcasson** [comment]
Yes, the least work for us it the ability to touch a file

**Comment by fcasson** [comment]
If I send my usual job termination signal, the running process will exit shortly after and then this will look like any other completed job to prominence.  So this case doesn't require the same special handling as a deleted or failed jobs (#84).

**Comment by alahiff** [comment]
In https://github.com/prominence-eosc/prominence/commit/53457163cafdf7486d22afa4c072aaf724755fce we catch SIGINT & SIGTERM sent by HTCondor to the first child of the starter. The running job is killed in the situation and we then carry out stage-out as normal (previously the stage-out step would never have happened).

Unfortunately in testing have found that if HTCondor cgroups are enabled, HTCondor doesn't appear to send SIGTERM 30s before killing the job - it juts wipes out all processes in the cgroup instantaneously. But when cgroups are disabled the new code works and output data from a deleted job can be downloaded. Will investigate to see if there's a way around this...

**Comment by fcasson** [comment]
`prominence exec` now meets this need for our use case, so from our point of view this issue is solved
